src/test/java/org/fungover/thunder/PackageReaderTest.java

```diff
@@ -85,38 +86,78 @@ void socketReceiveNoResponseIfNoValidConnectMessageIsSent() throws IOException {
         assertEquals(0, bytesWritten.length);
     }
     @Test
-    @DisplayName("Should return true when client send disconnect package")
-    void shouldReturnTrueWhenClientSendDisconnectPackage() throws IOException{
+    @DisplayName("Should return false if double connect message is received")
+    void shouldReturnFalsseIfDoubleConnectMessageIsReceived() throws IOException {
         Socket socketMock = mock(Socket.class);
-        setUpConnection(socketMock);
-        InputStream inputStream = new ByteArrayInputStream(new byte[]{(byte) 0xE0,0x00});
+        InputStream inputStream = new ByteArrayInputStream(new byte[]{0x10, 0x01, 0x00});
+        OutputStream outputStream = new ByteArrayOutputStream();
         when(socketMock.getInputStream()).thenReturn(inputStream);
```

Comment:
It looks like you have removed tests for isCleanDisconnect()-method, so the method does not have full coverage anymore.
![image](https://github.com/fungover/thunder/assets/143023413/d076b4e7-9bde-423e-b746-5a3630923ca8)

Reply:
Although the tests initially didn't work with the new code, we have managed to make the code functional with the tests again and have re-added the tests. Please feel free to leave a new review.